app/app.rb

```diff
@@ -0,0 +1,78 @@
+ENV['RACK_ENV'] ||= 'development'
+require_relative 'data_mapper_setup'
+require 'sinatra/base'
+require 'sinatra/flash'
+require 'bcrypt'
+
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+  enable :sessions
+  set :session_secret, 'super_secret'
+
+  register Sinatra::Flash
+
+  attr_reader :name
+
+  helpers do
+    def current_user
+      @current_user ||=User.get(session[:user_id])
+    end
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/sign_up' do
+    erb :sign_up
+  end
+
+  get '/chitters' do
+    erb :chitters
+  end
+
+  get '/peeps' do
+    @peeps = Peep.all(order: :created_at.desc)
+    erb :peeps
+  end
+
+  post '/peeps' do
+    user = User.get(session[:user_id])
+    @peep = user.peeps.create(message: params[:message], username: user.username)
+    redirect to('/peeps')
+  end
+
+  post '/chitters' do
+    @user = User.new(name: params[:name], username: params[:username], email: params[:email], password_digest: params[:password], password_confirmation: params[:password_confirmation])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect to('/chitters')
+    else
+      flash.now[:notices] = @user.errors.full_messages
+      erb :sign_up
+    end
+  end
```

Comment:
I'm confused by the URL resource naming here. Normally, a resource in the URL (here, 'chitters') will match the resource you're creating on the server ('users').